yarl/_url.py

```diff
@@ -1199,7 +1218,45 @@ def join(self, url: "URL") -> "URL":
         # See docs for urllib.parse.urljoin
         if not isinstance(url, URL):
             raise TypeError("url should be URL")
-        return URL(urljoin(str(self), str(url)), encoded=True)
+        other: URL = url
```

Comment:
```suggestion
        other = url
```

hint should be unneeded as its inferred from the isinstance